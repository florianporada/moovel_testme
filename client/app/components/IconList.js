import React from 'react';
import { View, Linking } from 'react-native';
import { Icon } from 'react-native-elements';
import PropTypes from 'prop-types';

import helper from '../helper';
import { styles, colors } from '../config/styles';

class IconList extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { modalContent } = this.props;

    return (
      <View style={[styles.profileCardGeneric, styles.profileCardIconWrap]}>
        <Icon
          raised
          name='send'
          type='font-awesome'
          color={colors.primary1}
          onPress={() => {
            if (helper.isEmpty(modalContent.email)) {
              helper.infoAlert();
            } else {
              Linking.openURL(`mailto:${modalContent.email}`);
            }
          }}
        />
        <Icon
          raised
          name='github'
          type='font-awesome'
          color={colors.primary1}
          onPress={() => {
            if (helper.isEmpty(modalContent.html_url)) {
              helper.infoAlert();
            } else {
              Linking.openURL(modalContent.html_url);
            }
          }}
        />
        <Icon
          raised
          name='globe'
          type='font-awesome'
          color={colors.primary1}
          onPress={() => {
            if (helper.isEmpty(modalContent.blog)) {
              helper.infoAlert();
            } else {
              Linking.openURL(modalContent.blog);
            }
          }}
        />
      </View>
    );
  }
}

IconList.propTypes = {
  modalContent: PropTypes.object
};

export default IconList;
