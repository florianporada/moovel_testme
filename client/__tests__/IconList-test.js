import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import IconList from '../app/components/IconList';


it('renders correctly', () => {
  const tree = renderer.create(
    <IconList />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
