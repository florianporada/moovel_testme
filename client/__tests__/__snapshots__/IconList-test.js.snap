// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 0,
        "flexDirection": "row",
        "paddingBottom": 15,
      },
      Object {
        "justifyContent": "space-between",
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Array [
          Object {
            "margin": 7,
          },
          Object {
            "borderRadius": 28,
            "height": 52,
            "width": 52,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "justifyContent": "center",
          },
          undefined,
        ],
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#4d86f7",
            "fontSize": 24,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Array [
          Object {
            "margin": 7,
          },
          Object {
            "borderRadius": 28,
            "height": 52,
            "width": 52,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "justifyContent": "center",
          },
          undefined,
        ],
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#4d86f7",
            "fontSize": 24,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Array [
          Object {
            "margin": 7,
          },
          Object {
            "borderRadius": 28,
            "height": 52,
            "width": 52,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "justifyContent": "center",
          },
          undefined,
        ],
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#4d86f7",
            "fontSize": 24,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
